python/bigipconfigdriver.py

```diff
@@ -609,51 +845,60 @@ def _do_reset(self):
                         self._backoff_time = 1
                         if self._backoff_timer is not None:
                             self.cleanup_backoff()
 
-                        # Manually delete custom profiles (if needed)
-                        if customProfiles:
-                            _delete_client_ssl_profiles(self._bigip,
-                                                        config['resources'])
-
-                        perf_enable = os.environ.get('SCALE_PERF_ENABLE')
-                        if perf_enable:  # pragma: no cover
-                            test_data = {}
-                            app_count = 0
-                            backend_count = 0
-                            for service in config['resources'][
-                                    'virtualServers']:
-                                app_count += 1
-                                backends = 0
-                                for pool in config['resources']['pools']:
-                                    if pool['name'] == service['name']:
-                                        backends = len(pool['poolMemberAddrs'])
-                                        break
-                                test_data[service['name']] = backends
-                                backend_count += backends
-                            test_data['Total_Services'] = app_count
-                            test_data['Total_Backends'] = backend_count
-                            test_data['Time'] = time.time()
-                            json_data = json.dumps(test_data)
-                            log.info('SCALE_PERF: Test data: %s', json_data)
+                    perf_enable = os.environ.get('SCALE_PERF_ENABLE')
+                    if perf_enable:  # pragma: no cover
+                        test_data = {}
+                        app_count = 0
+                        backend_count = 0
+                        for service in config['resources']['virtualServers']:
+                            app_count += 1
+                            backends = 0
+                            for pool in config['resources']['pools']:
+                                if pool['name'] == service['name']:
+                                    backends = len(pool['poolMemberAddrs'])
+                                    break
+                            test_data[service['name']] = backends
+                            backend_count += backends
+                        test_data['Total_Services'] = app_count
+                        test_data['Total_Backends'] = backend_count
+                        test_data['Time'] = time.time()
+                        json_data = json.dumps(test_data)
+                        log.info('SCALE_PERF: Test data: %s',
+                                 json_data)
 
                     log.debug('updating tasks finished, took %s seconds',
                               time.time() - start_time)
+
                 except Exception:
```

Comment:
This is scary.  I think you shouldn't catch all exceptions, you should catch expected runtime exceptions (and handle them).